src/components/WalletSelector/WalletButton.tsx

```diff
@@ -1,50 +1,68 @@
 import { Web3ReactManagerFunctions } from '@web3-react/core/dist/types';
 import { injected } from 'connectors/index';
-import { useCallback } from 'react';
+import { useCallback, useState } from 'react';
 import styled from 'styled-components';
+import { AbstractConnector } from '@web3-react/abstract-connector';
+import { useAuthActions, useAuthState } from 'contexts/auth/AuthContext';
+import { isLoadingState } from 'contexts/auth/types';
+import colors from 'components/core/colors';
 
 type WalletButtonProps = {
   walletName: string;
   activate: Web3ReactManagerFunctions['activate'];
-  connector: any;
+  connector: AbstractConnector;
 };
 
 function WalletButton({ walletName, activate, connector }: WalletButtonProps) {
+  const { setStateToLoading } = useAuthActions();
+  const [isConnecting, setIsConnecting] = useState(false);
+  const authState = useAuthState();
+  const isLoadingAuth = isLoadingState(authState);
+
   const handleClick = useCallback(() => {
+    setStateToLoading();
```

Comment:
hmm what will this accomplish? my worry is that setting to loading will throw the entire app into a loading state- like "APP IS LOADING"

Reply:
wondering if we can accomplish this with just the local `isConnecting` state?

Reply:
Good point, I wasn't sure what we wanted the Auth Loading state to represent, but I think I have a better idea now.

I've replied on the related comment in WalletSelector.tsx